include/picongpu/particles/Particles.tpp

```diff
@@ -217,6 +220,34 @@ namespace picongpu
 
         size_t sizeOfExchanges = 0u;
 
+        //std::cout << "Frame:\n\tSize: " << sizeof(FrameType) << "\n\tLLAMA view offset: " << offsetof(FrameType, view)
+        //          << "\n\tLLAMA view size: " << sizeof(FrameType::view) << std::endl;
+
+        {
+            //using View = decltype(FrameType::view);
+            using M = typename decltype(FrameType::view)::Mapping;
+            auto m = M{};
+            //View view;
+            //std::cout << "view begin " << &view << "\n";
+            //std::cout << "storage begin " << &view.storageBlobs[0][0] << "\n";
+            //for(auto ad : llama::ArrayDimsIndexRange{m.arrayDims()})
+            //{
+            //    llama::forEachLeaf<typename M::RecordDim>(
+            //        [&](auto coord)
+            //        {
+            //            auto& e = view(ad)(coord);
+            //            std::cout << "ad " << ad << " coord " << coord << " addr " << (void*) &e << "\n";
+            //        });
+            //}
+            //std::cout << "view end " << (&view + 1) << "\n";
+
+            std::ofstream{"llama_frame.html"} << llama::toHtml(m);
+            std::ofstream{"llama_frame.svg"} << llama::toSvg(m);
+
+            //PMACC_VERIFY(
+            //    reinterpret_cast<std::byte*>(&view) == reinterpret_cast<std::byte*>(&view.storageBlobs[0][0]));
+        }
+
```

Comment:
I initially used this for some debugging, but it is still a nice visualization of the frame memory layout. Should we keep it somehow? Maybe hidden behind a compilation option? Or dump it always?

Reply:
maybe separate to function which is debug compile only?

Reply:
So you want to have the memory layout dumps in debug builds? Fine for me!

Reply:
Any decision here? I'd like to keep the dumping code, but maybe behind a switch.

Reply:
Let's continue here: https://github.com/ComputationalRadiationPhysics/picongpu/pull/4003#pullrequestreview-1201027081